pkg/controller/custom_resource_replicas.go

```diff
@@ -0,0 +1,96 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	scaleclient "k8s.io/client-go/scale"
+
+	"github.com/grafana/rollout-operator/pkg/config"
+)
+
+func getCustomScaleResourceForStatefulset(ctx context.Context, sts *appsv1.StatefulSet, restMapper meta.RESTMapper, scalesGetter scaleclient.ScalesGetter) (*autoscalingv1.Scale, schema.GroupVersionResource, string, error) {
+	annotations := sts.GetAnnotations()
+	name := annotations[config.RolloutMirrorReplicasFromResourceNameAnnotationKey]
+	kind := annotations[config.RolloutMirrorReplicasFromResourceKindAnnotationKey]
+	if name == "" || kind == "" {
+		return nil, schema.GroupVersionResource{}, "", nil
+	}
+
+	apiVersion := annotations[config.RolloutMirrorReplicasFromResourceAPIVersionAnnotationKey]
+
+	reference := fmt.Sprintf("%s/%s/%s", kind, sts.Namespace, name)
```

Comment:
Maybe move this into the `if` statement where it's used below?

Reply:
I've now added use of this into another statement.